File: PokedexObjc/PokedexObjc/PokedexTableViewController.m
//
//  PokedexTableViewController.m
//  PokedexObjc
//
//  Created by Dillon McElhinney on 3/8/19.
//  Copyright © 2019 Dillon McElhinney. All rights reserved.
//

#import "PokedexTableViewController.h"
#import "DRMPokemon.h"
#import "PokedexObjc-Swift.h"

@interface PokedexTableViewController ()

@property NSArray<DRMPokemon *> *pokemons;

@end

@implementation PokedexTableViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.pokemons = [NSArray array];
    
    [PokemonAPI.shared fetchAllPokemonWithCompletion:^(NSArray<DRMPokemon *> * _Nullable pokemons, NSError * _Nullable error) {
        self.pokemons = pokemons;
        dispatch_sync(dispatch_get_main_queue(), ^{
            [self.tableView reloadData];
        });
    }];
}

#pragma mark - Table view data source
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.pokemons.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"PokemonCell" forIndexPath:indexPath];
    DRMPokemon *pokemon = self.pokemons[indexPath.row];
    
    [cell.textLabel setText:[pokemon.name capitalizedString]];
    
    return cell;
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: PokemonObjC/PokemonObjC/View Controllers/PokedexTableViewController.swift
//
//  PokedexTableViewController.swift
//  PokemonObjC
//
//  Created by Chris Gonzales on 4/24/20.
//  Copyright © 2020 Chris Gonzales. All rights reserved.
//

import UIKit

class PokedexTableViewController: UITableViewController {
    
    // MARK: - Properties
    
    private var pokemonArray: [CDGPokemon] = []
    
    // MARK: - View Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        fetchPokemon()
    }
    
    // MARK: - Private Methods
    
    private func fetchPokemon() {
        PokemonAPI.shared.fetchAllPokemon { (pokemonArray, error) in
            if let error = error {
             NSLog("Error fetching pokemon: \(error)")
                return
            }
            
            guard let pokemon = pokemonArray else {
                NSLog("No pokemon")
                return
            }
            self.pokemonArray.append(contentsOf: pokemon)
            DispatchQueue.main.async {
                self.tableView.reloadData()
            }
        }
    }

    // MARK: - Table view data source

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return 0
    }

    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "reuseIdentifier", for: indexPath)


        return cell
    }
  
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
}
